From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cem Mergenci <cmergenci@gmail.com>
Date: Thu, 9 Jan 2025 11:21:42 +0300
Subject: [PATCH] Implement a public setter for client accountID.

Required in order to address changes introduced by:
https://github.com/hashicorp/terraform-provider-aws/pull/40335/

Specifically this commit in the PR linked above:
https://github.com/hashicorp/terraform-provider-aws/commit/fa813c54d1645bda3135794c67ec34ee59deb3f4

Signed-off-by: Cem Mergenci <cmergenci@gmail.com>

diff --git a/internal/conns/awsclient_xp.go b/internal/conns/awsclient_xp.go
index b796849631..ed9e6a3055 100644
--- a/internal/conns/awsclient_xp.go
+++ b/internal/conns/awsclient_xp.go
@@ -18,3 +18,8 @@ func (c *AWSClient) AppendAPIOptions(options ...func(stack *middleware.Stack) er
 func (c *AWSClient) Session() *session_sdkv1.Session {
 	return c.session
 }
+
+// SetAccountID sets accountID of this client.
+func (c *AWSClient) SetAccountID(accountID string) {
+	c.accountID = accountID
+}
